Dockerfile

```diff
@@ -1,22 +1,48 @@
-FROM comicrelief/php7-slim:7.1
+FROM crs4/php7-base:7.1
 
 LABEL Description="Metadata backend for the PhenoMeNal Portal"
 LABEL software="PhenoMeNal Portal"
 LABEL software.version="1.1.2"
 LABEL version="0.2.2"
 
-RUN apt-get update && apt-get install -y --no-install-recommends mysql-client
-RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" && \
-    php -r "if (hash_file('SHA384', 'composer-setup.php') === '544e09ee996cdf60ece3804abc52599c22b1f40f4323403c44d44fdfdd586475ca9813a858088ffbc1f233e9b180f061') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;" && \
+# Optional arguments to choose the Git repo & branch to use at build time
+ARG git_repo=phnmnl/portal-metadata-backend
+ARG git_branch=master
+
+# Install required packages
+RUN apt-get update && apt-get install -y --no-install-recommends mysql-client python2.7 python-pip && \
+    pip install --upgrade oauth2client && \
+    php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" && \
     php composer-setup.php && \
-    php -r "unlink('composer-setup.php');"
-RUN apt-get -y install python2.7 && apt-get -y install python-pip
-RUN pip install --upgrade oauth2client
-RUN git clone --depth 1 --single-branch --branch master https://github.com/phnmnl/portal-metadata-backend.git 
+    php -r "unlink('composer-setup.php');" && \
+    rm -rf /var/lib/apt/lists/*
+
+# Download the PortalMetadataBackend
+RUN echo "Cloning branch '${git_branch}' of the Git repository '${git_repo}'" >&2 && \
+    git clone --depth 1 --single-branch --branch ${git_branch} https://github.com/${git_repo}.git
+
+# Set working directory
 WORKDIR portal-metadata-backend
-RUN php composer.phar install
-RUN chmod a+x setup_connection.sh
-ENV PATH /var/www/html/portal-metadata-backend/vendor/propel/propel/bin:$PATH
 
-# php -S localhost:8080 -t public public/index.php
+# Install backend dependencies
+RUN php composer.phar install && chmod a+x setup.sh
+
+# Set environment
+ENV MYSQL_HOST "localhost"
+ENV MYSQL_PORT 3306
+ENV SERVICE_PATH "/var/www/html/portal-metadata-backend"
+ENV PATH ${SERVICE_PATH}/vendor/propel/propel/bin:${PATH}
+
+# Add utility to wait for MySQL service
```

Comment:
This waiting logic really should be at the container orchestrator level (which I think we have), but having it here additionally doesn't hurt. But for our deployment, I would like the db waiting logic to be handled by Kubernetes through the init container.

Reply:
Yes, @pcm32! The waiting logic is mainly intended to instantiate the container without Kubernetes (e.g., with Docker Compose, etc.) and it is optional. Thus for our deployment we can use the db waiting logic handled by Kubernetes through the init container.